### DISCUSSION TITLE
Having Trouble With Nested Routes in Express

### DISCUSSION THREAD

**Discussion by josejaviersanahuja** [trpc/trpc#4018] [Q&A]
I was following the documentation but I can't implement nested routes at all. MergedRoutes are ok.

```typescript
// Init trpc
const t = initTRPC.context<Context>().create();

// router
export const router = t.router;

// Create Middleware
export const middleware = t.middleware;

// Create a procedure middleware
export const publicProcedure = t.procedure;

export const mergeRouters = t.mergeRouters;
```
```typescript
export const appRouter = router({
  user: usersRouter,
  activity: activitiesRouter,
  allData: allDataRouter,
});

export type AppRouter = typeof appRouter;
```
At the client I can see the next picture
![image](https://user-images.githubusercontent.com/82359964/226114441-735bcdb6-eecd-4a79-9e68-8d36191ec638.png)

It should look more like this:
![image](https://user-images.githubusercontent.com/82359964/226114513-dc0710b0-caa8-474b-81b9-f8e80fbaeb12.png)

I've never been able to implement nested routes. Please assist

**Comment by Nick-Lucas** [comment]
Are you declaring you AppRouter in the same file as your trpc instance?, you will get a circular dependency that way. Split them up like in the docs.

If that’s not the cause could you give some more information including the router code?

**Reply by josejaviersanahuja** [reply]
The answer is no. They are not in the same file.
```typescript
import { publicProcedure, router, mergeRouters } from "./trpc";
import { z } from "zod";
import debug from "debug";
import { allDataRouter } from "./component/allData/trpcRoute";
import { activitiesRouter } from "./component/activities/trcpRoute";
import { usersRouter } from "./component/users/trpcRoute";

const debugLog = debug("server:trpcRouter");
debugLog.enabled = true;

export const appRouter = router({
  hello: publicProcedure
    .input(
      z.object({
        name: z.string(),
      })
    )
    .query((req) => {
      return `Hello ${req.input.name}`;
    }),
  logToServer: publicProcedure.input(z.string()).mutation((req) => {
    debugLog(req.input);
  }),
  
  allData: allDataRouter,
  activities: activitiesRouter,
  users: usersRouter,
});

export type AppRouter = typeof appRouter;
```

I also opend a new issue, it may be related to this? It seems different but even more troublesome. rawInputs undefiend on mutate/mutations

https://github.com/trpc/trpc/issues/4020

**Reply by Nick-Lucas** [reply]
Got it, could you share your activities router? Useful to see the definition of that too

**Reply by Nick-Lucas** [reply]
If your repo is set up like the one in your reproduction for #4020, your issue is likely that you don't have a monorepo, you just have two apps in a repo together. Typescript likely can't find the types of the API because they're sitting in a different project with its own node_modules. If you use npm workspaces then packages can be hoisted to the root and discoverable by both projects, and this is enough for most people to make the client-side code discover types

**Reply by josejaviersanahuja** [reply]
Thank you very much for finding the problem on the issue I raised. 

Even though, the types at the client reaches when I don't use nested Routes. When I merge routes or add them in the main router, They reach perfectly to the front.

I will reorganize everything and try again. Thank you very much
